server/src/main/java/io/crate/types/ArrayType.java

```diff
@@ -242,6 +250,12 @@ private static <T> List<T> convert(@Nullable Object value,
         }
         if (value instanceof Collection<?> values) {
             return Lists.map(values, convertInner);
+        } else if (value instanceof byte[] b) {
+            try (StreamInput in = new ByteBufferStreamInput(ByteBuffer.wrap(b))) {
+                return in.readList(input -> (T) input.readGenericValue());
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
```

Comment:
So far decoding of the stored format was mostly contained in the column reference implementations. It looks we're now more conflating casts & internal storage - not sure if that's a good idea.

I'd tend to have a `decodeValue` or something like that instead on the `StorageSupport` or something like that. To have some write/read symmetry.
Basically an analog to the ValueIndexer